### PR TITLE
Review feature

### PR DESCRIPTION
**Pull request by DenysLiubchenko**
## Changes
- Added create review endpoint.
-  Added get all reviews by product id endpoint.
-  Added delete review endpoint.

### CODE REVIEW

File: dao/src/main/java/io/teamchallenge/repository/UserRepository.java [hunk 1]
```diff
@@ -44,4 +44,18 @@ public interface UserRepository extends JpaRepository<User, Long> {
...
     boolean existsByPhoneNumber(String phoneNumber);
+
+    /**
+     * Retrieves boolean true if there is already user with completed order with product.
+     *
+     * @param userId the phoneNumber of the user.
+     * @param productId the phoneNumber of the user.
+     * @return a boolean true if there is already user with completed order with product.
+     */
+    @Query("select case when count(u)> 0 then true else false end "
```
Comment (lagysha) [inline dao/src/main/java/io/teamchallenge/repository/UserRepository.java]: Can you simplify this with just count(u)> 0?

File: service/src/main/java/io/teamchallenge/service/ReviewService.java [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+     * @param addReviewRequestDto the data transfer object containing the details of the review to be added.
+     * @return a {@link ReviewResponseDto} representing the newly created review.
+     */
+    @Transactional
+    public ReviewResponseDto create(ReviewId reviewId, AddReviewRequestDto addReviewRequestDto) {
+        if (!userRepository.existsByIdAndCompletedOrderWithProductId(reviewId.getUserId(), reviewId.getProductId())) {
+            throw new ForbiddenException(USER_HAS_NO_COMPLETED_ORDERS_WITH_PRODUCT.formatted(reviewId.getProductId()));
+        }
+
+        User userReference = userRepository.getReferenceById(reviewId.getUserId());
```
Comment (lagysha) [inline service/src/main/java/io/teamchallenge/service/ReviewService.java:73]: Here can be a non-existing id, so you need to handle exception

File: service/src/main/java/io/teamchallenge/service/ReviewService.java [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+
+    /**
+     * Creates a new review for a product.
+     *
+     * @param reviewId            the composite ID of the review, which includes user ID and product ID.
+     * @param addReviewRequestDto the data transfer object containing the details of the review to be added.
+     * @return a {@link ReviewResponseDto} representing the newly created review.
+     */
+    @Transactional
+    public ReviewResponseDto create(ReviewId reviewId, AddReviewRequestDto addReviewRequestDto) {
```
Comment (lagysha) [inline service/src/main/java/io/teamchallenge/service/ReviewService.java:64]: Can a user make lots of comments on one product?

**Review by lagysha — CHANGES_REQUESTED** [review]

File: service/src/main/java/io/teamchallenge/service/ReviewService.java (hunk 3 above)
Comment (DenysLiubchenko) [reply service/src/main/java/io/teamchallenge/service/ReviewService.java:64]: No, only one

File: service/src/main/java/io/teamchallenge/service/ReviewService.java (hunk 2 above)
Comment (DenysLiubchenko) [reply service/src/main/java/io/teamchallenge/service/ReviewService.java:73]: I dont think so, id is got from token, therefore it must be existing. But I can handle EntityNotFoundException.

File: service/src/main/java/io/teamchallenge/service/ReviewService.java (hunk 3 above)
Comment (lagysha) [reply service/src/main/java/io/teamchallenge/service/ReviewService.java:64]: So, do that in code

**Review by lagysha — APPROVED** [review]
LGTM

**Pull request merged**